### PR TITLE
Fix authorization keyRoles

### PR DESCRIPTION
**Pull request by damianxu88**

### CODE REVIEW

**Comment by damianxu88** [comment]
@schowsf @realstraw Please review this PR

**Review by schowsf — APPROVED** [review]
LGTM

**Comment by realstraw** [comment]
looks like the env in https://github.com/salesforce/orchard/blob/master/orchard-ws/conf/application.conf#L22-L23 should be fixed too @damianxu88

**Comment by damianxu88** [comment]
@realstraw the application.conf format I put in last time that was merged should have had the changes in this PR applied to make it work, so this PR is to fix that

File: orchard-ws/app/utils/AuthorizationSettings.scala [hunk 1]
```diff
@@ -21,13 +21,12 @@ class AuthorizationSettings private (config: Config) {
     .entrySet()
     .asScala
     .filter(_.getKey.startsWith(s"hashed-keys."))
```
Comment (realstraw) [inline orchard-ws/app/utils/AuthorizationSettings.scala:23]: can just do `config.getConfig("hashed-keys")`?

File: orchard-ws/app/utils/AuthorizationSettings.scala [hunk 2]
```diff
@@ -21,13 +21,12 @@ class AuthorizationSettings private (config: Config) {
...
+    .foldLeft(Map[String, List[String]]()) { (b, kv) =>
+      val k: String = kv.getKey.stripPrefix(s"hashed-keys.")
       val configList: ConfigList = kv.getValue.asInstanceOf[ConfigList]
-      val listValue = configList.unwrapped().asScala.map(_.toString).toList
-      (key, listValue)
-    })
-    .toMap
+      val v = configList.unwrapped().asScala.map(_.toString).toList
+      (b -- v) ++ v.map(j => (j, b.getOrElse(j, List.empty) ++ List(k))).toMap
+    }
```
Comment (realstraw) [inline orchard-ws/app/utils/AuthorizationSettings.scala:29]: also this method would be a lot more readable if we implement it like this:

```
val userRoles = for {
  (r, us) <- config.getConfig("hashed-keys").root().asScala.toSeq
  u <- asInstanceOf[ConfigList].unwrapped().asScala
} yield u -> r

userRoles.groupBy(_._1).view.mapValues(_.map(_._2)).toMap
```

Comment (damianxu88) [reply orchard-ws/app/utils/AuthorizationSettings.scala:29]: @realstraw looks good, let me change

Comment (damianxu88) [reply orchard-ws/app/utils/AuthorizationSettings.scala]: @realstraw changed

File: orchard-ws/app/utils/AuthorizationSettings.scala (hunk 1 above)
Comment (damianxu88) [reply orchard-ws/app/utils/AuthorizationSettings.scala]: @realstraw changed

File: orchard-ws/app/utils/AuthorizationSettings.scala [hunk 3]
```diff
@@ -17,17 +17,14 @@ class AuthorizationSettings private (config: Config) {
...
-      val key: String = kv.getKey.stripPrefix(s"hashed-keys.")
-      val configList: ConfigList = kv.getValue.asInstanceOf[ConfigList]
-      val listValue = configList.unwrapped().asScala.map(_.toString).toList
-      (key, listValue)
-    })
-    .toMap
+  def keyRoles: Map[String, List[String]] = {
+    val userRoles = for {
+      (r, _) <- config.getConfig("hashed-keys").root().asScala.toList
+      u <- asInstanceOf[ConfigList].unwrapped().asScala
```
Comment (realstraw) [inline orchard-ws/app/utils/AuthorizationSettings.scala:23]: interesting, this is a new syntax I'm not aware

**Review by realstraw — APPROVED** [review]

**Pull request merged**